internal/shared/kubernetes/kubernetes.go

```diff
@@ -75,3 +83,33 @@ func GetSecret(ctx context.Context, client k8sClient.Client, namespace, secretNa
 	}
 	return secret, nil
 }
+
+// ApplyOwnershipReference adds  or updates ownerships to kubernetes resources in GitSync CRD
+func ApplyOwnershipReference(manifest string, gitSync *v1alpha1.GitSync) ([]byte, error) {
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	decoded, _, err := decode([]byte(manifest), nil, nil)
+	if err != nil {
+		return nil, err
+	}
+	obj, ok := decoded.(metav1.Object)
```

Comment:
Just to unify the code path and avoid errors here, I would recommend to use same pattern how we set [annotation](https://github.com/numaproj-labs/numaplane/blob/main/internal/shared/kubernetes/kubernetes.go#L37-L48), you can use `Unstructured.SetOwnerReferences`.

Reply:
I was looking at that. If we're going to do that, I wonder if we should also avoid unmarshaling yaml twice? What do you think @xdevxy ?

Reply:
I saw the new changes didn't take the suggestion, but I'm good with it as I don't feel strong about this.